typescript/src/schema/proto.ts

```diff
@@ -282,6 +283,50 @@ function pagingResponseType(
   return '.google.protobuf.FieldDescriptorProto.Type.' + type;
 }
 
+// Support DIREGAPIC google-discovery API pagination response with map field.
+function pagingMapResponseType(
+  messages: MessagesMap,
+  method: MethodDescriptorProto,
+  rest?: boolean
+) {
+  const pagingfield = pagingField(messages, method, undefined, rest);
+  const outputType = messages[method.outputType!];
+  if (!pagingfield || !pagingfield.type || !rest || !outputType.nestedType) {
+    return undefined;
+  }
+  const mapResponses = outputType.nestedType.filter(desProto => {
+    return desProto.options && desProto.options.mapEntry;
+  });
+  if (mapResponses.length === 0) {
+    return undefined;
+  }
+  if (mapResponses.length > 1) {
+    throw new Error(
+      `Paginated "${method.name}" method can only have one map field`
+    );
+  }
+  const pagingMapResponse = mapResponses[0];
+  if (pagingMapResponse.name !== pagingfield.typeName?.split('.').pop()) {
```

Comment:
Can you explain what are you checking here?  Maybe a comment would be useful here.

Reply:
// convert paging.field typeName .google.cloud.compute.v1.ItemEntry to ItemEntry
leave the comments.